### PR TITLE
Make Orientation respect the mask and make it a Predicate

### PR DESCRIPTION
**Pull request by ExE-Boss**

### CODE REVIEW

File: src/main/java/nova/core/component/transform/Orientation.java [hunk 1]
```diff
@@ -53,7 +55,7 @@
 	/**
 	 * The allowed rotation directions the block can face.
 	 */
-	public int rotationMask = 0x3C;
+	public Predicate<Direction> rotationMask = dir -> Arrays.stream(Direction.FLAT_DIRECTIONS).anyMatch(d -> d == dir);
```
Comment (RX14) [inline src/main/java/nova/core/component/transform/Orientation.java]: Also not a mask.

File: src/main/java/nova/core/component/transform/Orientation.java [hunk 2]
```diff
@@ -176,7 +206,8 @@ public Direction calculateDirection(Entity entity) {
 				.findFirst();
 
 			if (hit.isPresent()) {
-				return (isFlip) ? hit.get().side.opposite() : hit.get().side;
+				Direction dir = (isFlip) ? hit.get().side.opposite() : hit.get().side;
+				return rotationMask.test(dir) ? dir : calculateDirectionFromEntity(entity);
```
Comment (RX14) [inline src/main/java/nova/core/component/transform/Orientation.java]: Here's another version. I guess it's a tad cleaner.

```java
 	/**
 	 * Calculates the direction using raytracing
 	 * @param entity The entity to start raytracing from
 	 * @return The side of the block that is hit with the raytracing
 	 */
 	public Direction calculateDirection(Entity entity) {
 		if (provider instanceof Block) {
 			return new RayTracer(entity)
 				.setDistance(7)
 				.rayTraceBlocks(entity.world())
 				.filter(res -> res.block != provider)
  				.findFirst()
                .map(hit -> {
                    Direction dir = isFlip ? hit.side.opposite() : hit.side;
                    return rotationMask.test(dir) ? dir : calculateDirectionFromEntity(entity);
                })
                .orElse(Direction.UNKNOWN);
  		} else {  
 		    return Direction.UNKNOWN;
        }
 	}
```

File: src/main/java/nova/core/component/transform/Orientation.java [hunk 3]
```diff
@@ -199,19 +199,17 @@ public Orientation flipPlacement(boolean flip) {
 	 */
 	public Direction calculateDirection(Entity entity) {
 		if (provider instanceof Block) {
-			Optional<RayTracer.RayTraceBlockResult> hit = new RayTracer(entity)
-				.setDistance(7)
+ 			return new RayTracer(entity)
```
Comment (RX14) [inline src/main/java/nova/core/component/transform/Orientation.java]: I assume you tested my change?

**Comment by RX14** [comment]
Otherwise looks good to squash = merge

File: src/main/java/nova/core/component/transform/Orientation.java (hunk 3 above)
Comment (ExE-Boss) [inline src/main/java/nova/core/component/transform/Orientation.java]: Works

**Pull request merged**